From 2eca79896980bf2b40b915f2212ac71e7432f96c Mon Sep 17 00:00:00 2001
From: Qi Wang <interwq@gwu.edu>
Date: Wed, 7 Aug 2019 20:12:25 -0700
Subject: [PATCH 02/22] Fix tcaches mutex pre- / post-fork handling.

Signed-off-by: Pranav Vashi <neobuddy89@gmail.com>
Change-Id: I4c1e5241f1c500b90e22ce926138e45a72e413b3
---
 src/tcache.c | 12 +++---------
 1 file changed, 3 insertions(+), 9 deletions(-)

diff --git a/src/tcache.c b/src/tcache.c
index 9bb9ac23..4304794d 100644
--- a/src/tcache.c
+++ b/src/tcache.c
@@ -798,21 +798,15 @@ tcache_boot(tsdn_t *tsdn) {
 
 void
 tcache_prefork(tsdn_t *tsdn) {
-	if (!config_prof && opt_tcache) {
-		malloc_mutex_prefork(tsdn, &tcaches_mtx);
-	}
+	malloc_mutex_prefork(tsdn, &tcaches_mtx);
 }
 
 void
 tcache_postfork_parent(tsdn_t *tsdn) {
-	if (!config_prof && opt_tcache) {
-		malloc_mutex_postfork_parent(tsdn, &tcaches_mtx);
-	}
+	malloc_mutex_postfork_parent(tsdn, &tcaches_mtx);
 }
 
 void
 tcache_postfork_child(tsdn_t *tsdn) {
-	if (!config_prof && opt_tcache) {
-		malloc_mutex_postfork_child(tsdn, &tcaches_mtx);
-	}
+	malloc_mutex_postfork_child(tsdn, &tcaches_mtx);
 }
-- 
2.32.0

